cmd/tools/generate/generate.go

```diff
@@ -20,11 +29,55 @@ var systemdCmd = &cobra.Command{
 	Run:   doSystemd,
 }
 
+var dockerCmd = &cobra.Command{
+	Use:   "docker",
+	Short: "generate Harvest docker-compose.yml target for all pollers defined in config",
+	Run:   doDockerCompose,
+}
+
 func doSystemd(cmd *cobra.Command, _ []string) {
 	var config = cmd.Root().PersistentFlags().Lookup("config")
 	generateSystemd(config.Value.String())
 }
 
+func doDockerCompose(cmd *cobra.Command, _ []string) {
+	var config = cmd.Root().PersistentFlags().Lookup("config")
+	generateDockerCompose(config.Value.String())
+}
+
+func generateDockerCompose(path string) {
+	pollerTemplate := PollerTemplate{}
+	err := conf.LoadHarvestConfig(path)
+	if err != nil {
+		return
+	}
+	if conf.Config.Pollers == nil {
+		return
+	}
+	conf.IsDocker = true
+	for k, _ := range *conf.Config.Pollers {
+		port, _ := conf.GetPrometheusExporterPorts(k)
+		pollerTemplate.Pollers = append(pollerTemplate.Pollers, PollerPort{k, port})
+	}
+
+	t, err := template.New("docker-compose.tmpl").ParseFiles("docker/onePollerPerContainer/docker-compose.tmpl")
+	if err != nil {
+		panic(err)
+	}
+
+	color.DetectConsole("")
+	println("Docker Compose file generated at harvest home " + color.Colorize("docker-compose.yml", color.Green))
+	// Create the file
+	f, err := os.Create("docker-compose.yml")
```

Comment:
probably better to dump to stdout instead of overwriting file. simple enough for the operator to redirect to docker-compse.yml. You could mention that in your println

Reply:
done.